src/index.js

```diff
@@ -43,21 +56,49 @@ export const useSample = (sample) => {
       setPath([])
     }
   }, [paths])
-  // react to path or sample change
+  // find tabular data, react to path or sample change
+  const [data, setData] = useState([])
   useEffect(() => {
     if (path && path.length > 0 && sample) {
       setData(getByPath(sample, path))
     } else if (Array.isArray(sample) && sample.length > 0) {
       setData(sample)
     }
   }, [path, sample]) // in theory this reacts only to path change, while sample shouldn't change
+  // find data keys, react to data change
+  const [keys, setKeys] = useState([])
+  useEffect(() => {
+    try {
+      setKeys(buildKeys(data))
+    } catch(e) {
+      // TODO: handle error
+      setKeys([])
+    }
+  }, [data])
+  // set parsed keys by inferred value type, react to keys or data change
+  const [typedKeys, setTypedKeys] = useState({})
+  useEffect(() => {
+    if (keys && keys.length > 0 && data && data.length > 0) {
```

Comment:
or `if (keys?.length && data?.length)` 🌚

Reply:
revised in [c789f4f](https://github.com/EQWorks/api-explorer/pull/2/commits/c789f4fc2317940e14e22184e5d53e55402be738) along some other logic simplification/cleanup